teps/0046-pipelinerun-in-a-pod.md

```diff
@@ -0,0 +1,94 @@
+---
+status: proposed
+title: PipelineRun in a Pod
+creation-date: '2021-01-26'
+last-updated: '2021-01-26'
+authors:
+- '@jlpettersson'
+---
+
+# TEP-0046: PipelineRun in a Pod
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [Use Cases (optional)](#use-cases-optional)
+- [Requirements](#requirements)
+- [References (optional)](#references-optional)
+
+## Summary
+
+This TEP describes an alternative way to run a Pipeline composed of Tasks sharing data through Workspaces.
+
+The way we run Tasks in a Pipeline that share data through a Workspace today has several problems, some
+of them are related to the need to use Kubernetes Persistent Volumes to back Workspaces and to
+schedule the different TaskRun Pods in an appropriate way, especially to allow multiple of them to run
+concurrently while accessing the same Workspace volume.
+
+[#3638: Investigate if we can run whole PipelineRun in a Pod](https://github.com/tektoncd/pipeline/issues/3638)
+
+[Design doc: Task parallelism when using workspace](https://docs.google.com/document/d/1lIqFP1c3apFwCPEqO0Bq9j-XCDH5XRmq8EhYV_BPe9Y)
+
+## Motivation
+
+Currently the only way to run a Pipeline composed of Tasks is that each task is executed in its own Pod. If you combine
+Tasks in a Pipeline and they need to share data (beyond a simple
+[result](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#using-results)),
+you'll need to provision a PVC or do some other similar, cloud specific thing,
+to [make a volume available](https://github.com/tektoncd/pipeline/blob/master/docs/workspaces.md#specifying-volumesources-in-workspaces)
+that can be shared between them.
+
+Some problems with this:
+* PVCs add additional overhead, both in speed, in resource costs and in management.
+* PVCs is hard to share between multiple Pods concurrently - needs appropriate scheduling of the Pods.
+* PVCs is difficult to share in subsequently run Pods that might be scheduled to a different AZ than where the
+  volume is located.
+
+### Goals
+
+- Make it possible to run whole PipelineRun (including all TaskRuns) within a single Pod so that the workspace can be within the Pod:
+    - At runtime in a PipelineRun
```

Comment:
hmm interesting, this is a difference with https://github.com/tektoncd/community/pull/316